src/libraries/Microsoft.PowerFx.Core/Parser/PartialAttribute.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Parser
+{
+    internal class PartialAttribute
+    {
+        public enum AttributeOperationKind
+        {
+            Error,
+            PartialAnd,
+            PartialOr,
+            PartialTable,
+            PartialRecord
+        }
+
+        public readonly IdentToken AttributeName;
+
+        // This probably should be a TexlNode, but for now we're prototyping a simple case
+        // of just [Partial And]-type attributes, where they're pairs of name and operation.
+        public readonly Token AttributeOperationToken;
+
+        public readonly AttributeOperationKind AttributeOperation;
+
+        public PartialAttribute(IdentToken attributeName, Token attributeOperation)
+        {
+            AttributeName = attributeName;
+            AttributeOperationToken = attributeOperation;
+            AttributeOperation = ToKind(attributeOperation);
+        }
+
+        private AttributeOperationKind ToKind(Token attributeOperation)
+        {
+            if (attributeOperation is KeyToken keyTok)
+            {
+                if (keyTok.Kind == TokKind.KeyAnd)
+                {
+                    return AttributeOperationKind.PartialAnd;
+                }
+                else if (keyTok.Kind == TokKind.KeyOr)
+                {
+                    return AttributeOperationKind.PartialOr;
+                }
+            }
+            else if (attributeOperation is IdentToken identTok)
+            {
+                if (identTok.Name.Value == "Table")
+                {
+                    return AttributeOperationKind.PartialTable;
+                }
+                else if (identTok.Name.Value == "Record")
+                {
+                    return AttributeOperationKind.PartialRecord;
+                }
+            }
+
+            return AttributeOperationKind.Error;
+        }
+
+        public bool SameAttribute(PartialAttribute other)
```

Comment:
>SameAttribute [](http://example.com/codeflow?start=20&length=13)

Why not override Equals?

Reply:
It's not equality. The tokens are different, and would have different spans. We're just ensuring the ids match